### ISSUE TITLE
Figure out why parJoin is not as fast as it could be.

### ISSUE THREAD

**Issue by ValdemarGr** [ValdemarGr/gql#234]
Maybe a more efficient version of parJoin can be implemented if the stream's outputs are discarded. Resources shouldn't be an issue since the data is discarded (parJoin puts leases on the outer stream).
```scala
(streams: Stream[F, Stream[F, O]]) => 
    Stream.eval(F.deferred[Either[Throwable, Unit]]).flatMap{ d =>
      Stream.eval(SignallingRef.of(())).flatMap{ sigRef => // produces an initial element also! use option + unNone for other semantics
        Stream.eval(F.deferred[Unit]).flatMap{ c =>
          val bg = streams.flatMap(sub => Stream.supervise(
            sub
              .evalMap(_ => sigRef.set(()))
              .compile
              .drain
              .onError(d.complete(_))
              .onCancel(c.complete(()))
            ))

            sigRef.discrete.interruptWhen(d).interruptWhen(c.get >> F.cancelled).concurrently(bg)
        }
    }
  }
```

**Comment by ValdemarGr** [comment]
It seems to be over 10 times faster than `parJoinUnbounded`, this should go into the docs.

**Comment by ValdemarGr** [comment]
fixed in #236
